MediaAdmin/Reference/Strategies/MediaInNodeReferenceStrategy.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace OpenOrchestra\MediaAdmin\Reference\Strategies;
+
+use OpenOrchestra\ModelInterface\Model\ReadNodeInterface;
+use OpenOrchestra\Backoffice\Reference\Strategies\ReferenceStrategyInterface;
+use OpenOrchestra\ModelInterface\Model\NodeInterface;
+
+/**
+ * Class MediaInNodeReferenceStrategy
+ */
+class MediaInNodeReferenceStrategy extends AbstractMediaReferenceStrategy implements ReferenceStrategyInterface
+{
+    /**
+     * @param mixed $entity
+     *
+     * @return boolean
+     */
+    public function support($entity)
+    {
+        return ($entity instanceof ReadNodeInterface);
+    }
+
+    /**
+     * @param mixed $entity
+     */
+    public function addReferencesToEntity($entity)
+    {
+        if ($this->support($entity)) {
```

Comment:
@itkg-nanne : Yes but as discussed with @alavieille it's a fault that should be corrected on our other managers. According to the kind of strategy, giving an argument not supported can lead to something going from nothing particular to a fatal error like on these strategies.
Each method of a class (including the strategies ones) should always check by themselves that they support the parameters, they can't rely on an external source to do it.

Symfony voter are for instance coded that way.